engine/src/commonMain/kotlin/com/landry/digital/engine/component/LogicGate.kt

```diff
@@ -1,8 +1,11 @@
 package com.landry.digital.engine.component
 
+import com.landry.digital.engine.properties.GateProperties
+
 interface LogicGate {
     val inputs: MutableList<Pin>
     val outputs: List<Pin>
+    var gateProperties: GateProperties?
```

Comment:
A more abstracted GateProperties interface avoids forcing the user of 'engine' to think about UI (if I just have a CLI, or am handling the circuit in code, the idea of position may not be necessary). It even allows other interfaces to inject their own properties instead.